src/main/kotlin/react/bootstrap/components/button/Button.kt

```diff
@@ -0,0 +1,273 @@
+            class LIGHT internal constructor() : Variants(ClassNames.BTN_LIGHT)
+            class LINK internal constructor() : Variants(ClassNames.BTN_LINK)
+            class PRIMARY internal constructor() : Variants(ClassNames.BTN_PRIMARY)
+            class SECONDARY internal constructor() : Variants(ClassNames.BTN_SECONDARY)
+            class SUCCESS internal constructor() : Variants(ClassNames.BTN_SUCCESS)
+            class WARNING internal constructor() : Variants(ClassNames.BTN_WARNING)
+
+            companion object {
+                val DANGER = DANGER()
+                val DARK = DARK()
+                val INFO = INFO()
+                val LIGHT = LIGHT()
+                val LINK = LINK()
+                val PRIMARY = PRIMARY()
+                val SECONDARY = SECONDARY()
+                val SUCCESS = SUCCESS()
+                val WARNING = WARNING()
+            }
+        }
+
+        class Outline {
+            class DANGER internal constructor() : Variants(ClassNames.BTN_OUTLINE_DANGER)
+            class DARK internal constructor() : Variants(ClassNames.BTN_OUTLINE_DARK)
+            class INFO internal constructor() : Variants(ClassNames.BTN_OUTLINE_INFO)
+            class LIGHT internal constructor() : Variants(ClassNames.BTN_OUTLINE_LIGHT)
+            class PRIMARY internal constructor() : Variants(ClassNames.BTN_OUTLINE_PRIMARY)
+            class SECONDARY internal constructor() : Variants(ClassNames.BTN_OUTLINE_SECONDARY)
+            class SUCCESS internal constructor() : Variants(ClassNames.BTN_OUTLINE_SUCCESS)
+            class WARNING internal constructor() : Variants(ClassNames.BTN_OUTLINE_WARNING)
+
+            companion object {
+                val DANGER = DANGER()
+                val DARK = DARK()
+                val INFO = INFO()
+                val LIGHT = LIGHT()
+                val PRIMARY = PRIMARY()
+                val SECONDARY = SECONDARY()
+                val SUCCESS = SUCCESS()
+                val WARNING = WARNING()
+            }
+        }
+    }
+
+    enum class Sizes(override val className: ClassNames) : ClassNameEnum {
+        SM(ClassNames.BTN_SM),
+        LG(ClassNames.BTN_LG);
+    }
+
+    sealed class Types {
+        class Button(
+            val formEncType: ButtonFormEncType? = null,
+            val formMethod: ButtonFormMethod? = null,
+            val type: ButtonType = ButtonType.button
+        ) : Types()
+
+        class Input(
+            val type: Type = Type.BUTTON,
+            val formEncType: InputFormEncType? = null,
+            val formMethod: InputFormMethod? = null,
+            val name: String? = null,
+            val value: String
+        ) : Types() {
+            enum class Type(val inputType: InputType) {
+                BUTTON(InputType.button),
+                RESET(InputType.reset),
+                SUBMIT(InputType.submit);
+            }
+        }
+
+        class Link(
+            val href: String? = null,
```

Comment:
what about route links? This could be done through some fancy cloning. I could add another type which just receives a ReactElement

Reply:
later